### ISSUE TITLE
Exception in HealthCheckReportCollector.GetHealthCheckExecution

### ISSUE THREAD

**Issue by espenrl** [Xabaril/AspNetCore.Diagnostics.HealthChecks#171]
I don't see how SingleOrDefaultAsync ends up with multiple results in this case. Any ideas?

```
HealthCheck collector HostedService throw a error: Source sequence contains more than one element.
System.InvalidOperationException: Source sequence contains more than one element.
   at async Task<TSource> System.Linq.AsyncEnumerable.SingleOrDefault_<TSource>(IAsyncEnumerable<TSource> source, CancellationToken cancellationToken)
   at async Task<bool> Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider+TaskResultAsyncEnumerable<T>+Enumerator.MoveNext(CancellationToken cancellationToken)
   at async Task<bool> System.Linq.AsyncEnumerable+SelectEnumerableAsyncIterator<TSource, TResult>.MoveNextCore(CancellationToken cancellationToken)
   at async Task<bool> System.Linq.AsyncEnumerable+AsyncIterator<TSource>.MoveNext(CancellationToken cancellationToken)
   at async Task<bool> Microsoft.EntityFrameworkCore.Query.Internal.AsyncLinqOperatorProvider+ExceptionInterceptor<T>+EnumeratorExceptionInterceptor.MoveNext(CancellationToken cancellationToken)
   at async Task<TResult> Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteSingletonAsyncQuery<TResult>(QueryContext queryContext, Func<QueryContext, IAsyncEnumerable<TResult>> compiledQuery, IDiagnosticsLogger<Query> logger, Type contextType)
   at async Task<HealthCheckExecution> HealthChecks.UI.Core.HostedService.HealthCheckReportCollector.GetHealthCheckExecution(HealthCheckConfiguration configuration)
   at async Task<bool> HealthChecks.UI.Core.HostedService.HealthCheckReportCollector.HasLivenessRecoveredFromFailure(HealthCheckConfiguration configuration)
   at async Task HealthChecks.UI.Core.HostedService.HealthCheckReportCollector.Collect(CancellationToken cancellationToken)
   at async Task HealthChecks.UI.Core.HostedService.HealthCheckCollectorHostedService.ExecuteAsync(CancellationToken cancellationToken)
```

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/d650f3f2bc0944505959e5a9ce875c6d12a9ba46/src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs#L104-L110

![image](https://user-images.githubusercontent.com/4621581/56962298-7ffa9980-6b56-11e9-8728-4c1f4bb01909.png)

**Comment by OmerNassie** [comment]
Why is it closed? I have the same issue now on 3.1.1

**Comment by jaredthirsk** [comment]
Is #868 related?
